R/0_documentation.R

```diff
@@ -17,48 +62,95 @@ rd_source_conn <- function(type = "param") {
 
 rd_target_conn <- function(type = "param") {
   checkmate::assert_choice(type, c("param", "field"))
-  paste("target_conn (`DBIConnection`)\\cr",
+  paste("(`DBIConnection`)\\cr",
         "A database connection to store the computed features in.",
         ifelse(type == "field", " Read only.", ""))
 }
 
 
 rd_target_schema <- function(type = "param") {
   checkmate::assert_choice(type, c("param", "field"))
-  paste("target_schema (`character`)\\cr",
+  paste("(`character`)\\cr",
         "The schema to place the feature store in.",
         ifelse(type == "field", " Read only.", ""),
         "If the database backend does not support schema, the tables will be prefixed with target_schema.")
 }
 
 
+rd_training_length <- function(type = "param") {
+  checkmate::assert_choice(type, c("param", "field"))
+  paste("(`numeric`)\\cr",
+        "The number of days that should be included in the training of the model.",
+        ifelse(type == "field", " Read only.", ""))
+}
+
+
 rd_start_date <- function(type = "param") {
   checkmate::assert_choice(type, c("param", "field"))
-  paste("start_date (`Date`)\\cr",
+  paste("(`Date`)\\cr",
         "Study period start.",
         ifelse(type == "field", " Read only.", ""))
 }
 
 
 rd_slice_ts <- function(type = "param") {
   checkmate::assert_choice(type, c("param", "field"))
-  paste("slice_ts (`Date` or `character`)\\cr",
+  paste("(`Date` or `character`)\\cr",
         "Date to slice the database on (used if source_conn is a database).",
         ifelse(type == "field", " Read only.", ""))
 }
 
 
 rd_end_date <- function(type = "param") {
   checkmate::assert_choice(type, c("param", "field"))
-  paste("end_date (`Date`)\\cr",
+  paste("(`Date`)\\cr",
         "Study period end.",
         ifelse(type == "field", " Read only.", ""))
 }
 
 
-rd_.data <- function(type = "param") { # nolint: object_name_linter
+rd_.data <- function(type = "param") {                                                                                  # nolint: object_name_linter
```

Comment:
That's an awful lot of spaces

```suggestion
rd_.data <- function(type = "param") { # nolint: object_name_linter
```

Reply:
I have started moving the `# nolint:` statements are moved past the 120 character line.
I have a strong preference for this since
1) They are not "code comments" and thus should be visually distinct from "code comments"
2) They do not trigger lining issues by being past the 120 character line.

By placing these statements past the 120 character line, it clearly indicate that they are different.

Reply:
That does make sense.

I can't however help thinking about the user adding code comments to a line with one of these "meta-comments", unknowingly expanding the line. And since the statement disables the linter, they naturally go unnoticed by that.

I would additionally argue that `notlint` comments still _technically_ still are code comments; they are just not comments meant for humans, if that makes sense.